brainclouds2s.js

```diff
@@ -29,101 +27,68 @@ function s2sRequest(context, json, callback)
         }
     }
 
-    var req = https.request(options, res =>
-    {
+    var req = https.request(options, res => {
         var data = ''
-        
+
         // A chunk of data has been recieved.
-        res.on('data', chunk =>
-        {
+        res.on('data', chunk => {
             data += chunk
         })
-        
+
         // The whole response has been received. Print out the result.
-        res.on('end', () =>
-        {
-            if (context.logEnabled)
-            {
+        res.on('end', () => {
+            if (context.logEnabled) {
                 console.log(`[S2S RECV ${context.appId}] ${data}`)
             }
-            if (callback)
-            {
-                if (data)
-                {
-                    
-                    try {
-                        let dataJson = JSON.parse(data)
-                        callback(context, dataJson)
-                    } 
-                    catch (error) 
-                    {
-                        if (context.logEnabled)
-                        {
-                            console.log(`[S2S Error ${context.appId}] ${error}`)
-                        }
-                        if (callback)
-                        {
-                            callback(context, null)
-                        }
-                    }
-                    
+            let responseData = null;
+            if (data) {
+                try {
+                    responseData = JSON.parse(data);
                 }
-                else
-                {
-                    callback(context, null)
+                catch (error) {
+                    console.log(`[S2S Error ${context.appId}] ${error}`)
+                    console.log("Failed to parse response");
```

Comment:
I don't think we want this 2nd log message... just change the `[S2S Error ...` above to something like `[S2S Error parsing response ...`